### ISSUE TITLE
Have continuous tests for the iOS library

### ISSUE THREAD

**Issue by jcanizales** [grpc/grpc#2119]
Travis doesn't have Macs for us, but soon we'll have dedicated ones for our Jenkins instance. By then, we should have a script to:

1. Install cocoapods and homebrew (or we have them permanently installed?).
2. Install protoc and the gRPC plugins from head.
3. pod install in the Objective-C tests directory.
4. Compile and run locally a RouteGuide server.
5. xcodebuild test

**Comment by ctiller** [comment]
Recommend developing a way to test this that doesn't require root access to
help keep everything as hermetic as possible.

**Comment by jcanizales** [comment]
Notes.

To install Cocoapods locally: https://guides.cocoapods.org/using/getting-started.html#sudo-less-installation

To install Homebrew locally:
```bash
cd ~
mkdir homebrew && curl -L https://github.com/Homebrew/homebrew/tarball/master | tar xz --strip 1 -C homebrew
export PATH=$HOME/homebrew/bin:$PATH
```

To configure an XCode project properly for CI (plus troubleshooting): https://github.com/facebook/xctool#continuous-integration

**Comment by jcanizales** [comment]
Blocks https://github.com/grpc/grpc/issues/2127 and https://github.com/grpc/grpc/issues/2128

**Comment by jtattermusch** [comment]
I think we should have this for beta.

**Comment by ctiller** [comment]
s/should/must, in my view

**Comment by jcanizales** [comment]
Yes, as we decided yesterday, testing coverage is a requirement for Beta.

**Comment by nicolasnoble** [comment]
What I need from you is the list of softwares to install on MacOS, and a script that builds and run the tests.

**Comment by jcanizales** [comment]
Nico: software that needs to be installed in the Mac:
* XCode
* XCode's "Command Line Tools": From Xcode, install via _Xcode &rarr; Preferences &rarr; Downloads_.
* [Cocoapods](https://guides.cocoapods.org/using/getting-started.html#sudo-less-installation)

The current script I have in #2538 expects `protoc` to be installed and on the path. I can adjust it so that it compiles `protoc` from the submodule and runs the resulting binary. But I expect running a local gRPC server will require it "installed and on the path" anyway, right?

**Comment by jcanizales** [comment]
Also, note that that script *is not* what we should run on Jenkins. I'll update you when that's ready.

**Comment by jcanizales** [comment]
`./tools/run_tests/run_tests.py -l objc` can now be used by Jenkins.

It still expects `protoc` to be installed.

**Comment by jcanizales** [comment]
Ok, xcodebuild and cocoapods are installed in the Mac Jenkins machine. Now I only need to fix the build scripts to use `protoc` from `bins/$CONFIG/protobuf/` if it's not installed (it won't be installed in the machine).

**Comment by jcanizales** [comment]
The tests are still failing, and looking into it via ssh it seems XCode isn't installed on the machine :) (only XCode's command line tools are). Because installation is via iTunes, I can't do it myself via ssh, though.

**Comment by jcanizales** [comment]
This is blocked on issue #2807.

**Comment by jcanizales** [comment]
We're currently trying to have them on Travis. [They're failing there](https://travis-ci.org/nicolasnoble/grpc/builds/82222666), so I need to debug what's going on.

**Comment by jcanizales** [comment]
Fixed by #3998 :) :)
